buildhat/devicetypes.py

```diff
@@ -0,0 +1,70 @@
+class DeviceTypes:
+    """Translate BuildHAT device types"""
+
+    _device_names = {  2: "PassiveMotor",
+                       8: "Light",
+                      34: "TiltSensor",
+                      35: "MotionSensor",
+                      37: "ColorDistanceSensor",
+                      61: "ColorSensor",
+                      62: "DistanceSensor",
+                      63: "ForceSensor",
+                      64: "Matrix",
+                      38: "Motor",
+                      46: "Motor",
+                      47: "Motor",
+                      48: "Motor",
+                      49: "Motor",
+                      65: "Motor",
+                      75: "Motor",
+                      76: "Motor"
+                    }
+
+    # With corresponding part numbers
+    _device_descriptions = {  2: "PassiveMotor",
+                       8: "Light", #88005
+                      34: "WeDo 2.0 Tilt Sensor", #45305, not mentioned in docs anywhere
+                      35: "MotionSensor", # Maybe 45304
+                      37: "Color & Distance Sensor", #88007
+                      61: "Color Sensor", #45605
+                      62: "Distance Sensor", #45604
+                      63: "Force Sensor", #45606
+                      64: "3x3 Color Light Matrix", #45608
+                      38: "Medium Linear Motor", # 88008
+                      46: "Large Motor", #88013
+                      47: "XL Motor", #88014
+                      48: "Medium Angular Motor (Cyan)", #45603 RPi docs says this is the "Large Angular" (Cyan), which is an emperical mismatch against 45603
+                      49: "Motor", # BuildHAT serial protocol PDF says this is the "SPIKE Prime large motor" (which is Cyan).  RPi docs call id 31 "Medium Angular Motor"
+                      65: "Small Angular Motor", #45607
+                      75: "Medium Angular Motor", # Maybe 88018, came with 51515, which lists as 6299646, not 6359216
+                      76: "Motor"
```

Comment:
You might find [this](https://github.com/pybricks/technical-info/blob/master/assigned-numbers.md#io-device-type-ids) useful to solve some of the ?s here.

Reply:
Indeed.  The Mario codes are fascinating in that they're in the same space, but over bluetooth.